sdk/validatorManager/validator_manager.go

```diff
@@ -0,0 +1,221 @@
+	errInvalidWarpSourceChainID            = fmt.Errorf("invalid wapr source chain ID")
+	errInvalidWarpOriginSenderAddress      = fmt.Errorf("invalid warp origin sender address")
+	errorSignatureToError                  = map[string]error{
+		"InvalidInitialization()":                      ErrAlreadyInitialized,
+		"InvalidMaximumChurnPercentage(uint8)":         errInvalidMaximumChurnPercentage,
+		"InvalidValidationID(bytes32)":                 errInvalidValidationID,
+		"InvalidValidatorStatus(uint8)":                errInvalidValidatorStatus,
+		"MaxChurnRateExceeded(uint64)":                 errMaxChurnRateExceeded,
+		"InvalidInitializationStatus()":                errInvalidInitializationStatus,
+		"InvalidValidatorManagerBlockchainID(bytes32)": errInvalidValidatorManagerBlockchainID,
+		"InvalidValidatorManagerAddress(address)":      errInvalidValidatorManagerAddress,
+		"NodeAlreadyRegistered(bytes)":                 errNodeAlreadyRegistered,
+		"InvalidSubnetConversionID(bytes32,bytes32)":   errInvalidSubnetConversionID,
+		"InvalidRegistrationExpiry(uint64)":            errInvalidRegistrationExpiry,
+		"InvalidBLSKeyLength(uint256)":                 errInvalidBLSKeyLength,
+		"InvalidNodeID(bytes)":                         errInvalidNodeID,
+		"InvalidWarpMessage()":                         errInvalidWarpMessage,
+		"InvalidWarpSourceChainID(bytes32)":            errInvalidWarpSourceChainID,
+		"InvalidWarpOriginSenderAddress(address)":      errInvalidWarpOriginSenderAddress,
+	}
+)
+
+// PoAValidatorManagerInitialize initializes contract [managerAddress] at [rpcURL], to
+// manage validators on [subnetID], with
+// owner given by [ownerAddress]
+func PoAValidatorManagerInitialize(
+	rpcURL string,
+	managerAddress common.Address,
+	privateKey string,
+	subnetID ids.ID,
+	ownerAddress common.Address,
+) (*types.Transaction, *types.Receipt, error) {
+	const (
+		defaultChurnPeriodSeconds     = uint64(0)
+		defaultMaximumChurnPercentage = uint8(20)
+	)
+	type Params struct {
+		SubnetID               [32]byte
+		ChurnPeriodSeconds     uint64
+		MaximumChurnPercentage uint8
+	}
+	params := Params{
+		SubnetID:               subnetID,
+		ChurnPeriodSeconds:     defaultChurnPeriodSeconds,
+		MaximumChurnPercentage: defaultMaximumChurnPercentage,
+	}
+	return contract.TxToMethod(
+		rpcURL,
+		privateKey,
+		managerAddress,
+		nil,
+		"initialize PoA manager",
+		errorSignatureToError,
+		"initialize((bytes32,uint64,uint8),address)",
+		params,
+		ownerAddress,
+	)
+}
+
+// PoaValidatorManagerGetPChainSubnetConversionWarpMessage constructs p-chain-validated (signed) subnet conversion warp
+// message, to be sent to the validators manager when
+// initializing validators set
+// the message specifies [subnetID] that is being converted
+// together with the validator's manager [managerBlockchainID],
+// [managerAddress], and the initial list of [validators]
+func PoaValidatorManagerGetPChainSubnetConversionWarpMessage(
```

Comment:
We should be consistent with our capitalization of PoA. I think either PoA or POA makes sense.